src/util/system.cpp

```diff
@@ -656,15 +656,37 @@ fs::path GetDefaultDataDir()
     else
         pathRet = fs::path(pszHome);
 #ifdef MAC_OSX
-    // Mac
+    // macOS
     return pathRet / "Library/Application Support/Bitcoin";
 #else
-    // Unix
+    // Unix-like
     return pathRet / ".bitcoin";
 #endif
 #endif
 }
 
+namespace {
+fs::path StripLastElementOfPath(const fs::path& path)
+{
+    fs::path result;
+    for (auto it = path.begin(); it != --path.end(); ++it) {
+        result /= *it;
+    }
+    return result;
+}
+
+fs::path StripRedundantLastElementsOfPath(const fs::path& path)
+{
+    auto result = path;
+    while (result.filename().string() == ".") {
+        result = StripLastElementOfPath(result);
```

Comment:
nit: Can be written shorter

```suggestion
        result = result.parent_path();
```

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/20080#issuecomment-717499942).